tar/test/main.c

```diff
@@ -130,6 +130,13 @@ __FBSDID("$FreeBSD: src/usr.bin/tar/test/main.c,v 1.6 2008/11/05 06:40:53 kientz
 # include <crtdbg.h>
 #endif
 
+mode_t umasked(mode_t expected_mode)
+{
+	mode_t mode = umask(0);
+	umask(mode);
+	return expected_mode & ~mode;
+}
+
```

Comment:
Can you please elaborate?  I'm not sure what you mean by `2920`, I was about to use the idiom from `main.c`, but still that was 3-line function.